src/Infrastructure.Dapper/KeyManagement/Repositories/UserAsymmetricKeysRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿#nullable enable
+using System.Data;
+using Bit.Core.KeyManagement.Models.Data;
+using Bit.Core.KeyManagement.Repositories;
+using Bit.Core.Settings;
+using Bit.Infrastructure.Dapper.Repositories;
+using Dapper;
+using Microsoft.Data.SqlClient;
+
+namespace Bit.Infrastructure.Dapper.KeyManagement.Repositories;
+
+public class UserAsymmetricKeysRepository : BaseRepository, IUserAsymmetricKeysRepository
+{
+    public UserAsymmetricKeysRepository(GlobalSettings globalSettings)
+        : this(globalSettings.SqlServer.ConnectionString, globalSettings.SqlServer.ReadOnlyConnectionString)
+    {
+    }
+
+    public UserAsymmetricKeysRepository(string connectionString, string readOnlyConnectionString) : base(
+        connectionString, readOnlyConnectionString)
+    {
+    }
+
+    public async Task RegenerateUserAsymmetricKeysAsync(UserAsymmetricKeys userAsymmetricKeys)
+    {
+        await using var connection = new SqlConnection(ConnectionString);
+
+        await connection.ExecuteAsync("[dbo].[UserAsymmetricKeys_Regenerate]",
+            userAsymmetricKeys, commandType: CommandType.StoredProcedure);
```

Comment:
The last parameter of `UserAsymmetricKeys_Regenerate` is named `PrivateKey`, but there is no equivalently named field in `UserAsymmetricKeys`. There is `UserKeyEncryptedPrivateKey`.

Reply:
Great catch fixed in https://github.com/bitwarden/server/pull/4898/commits/9278677d3065435027ea97c22ba81266fd33e99a